pkg/blockbuilder/blockbuilder_metrics.go

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package blockbuilder
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type blockBuilderMetrics struct {
```

Comment:
I would remove all metrics not tracked for now. It's quite difficult to review metrics that are not currently used. You can introduce them in the actual PRs where these metrics are tracked.